sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FailureSafeParser.scala

```diff
@@ -56,9 +58,15 @@ class FailureSafeParser[IN](
     }
   }
 
+  private val skipParsing = !isMultiLine && mode == PermissiveMode && schema.isEmpty
```

Comment:
not a big deal but I would leave a comment to explain why it's permissive and non-miltiline only. I assume counts are known when it's actually parsed for multiline cases, and counts should be given in any case when the mode is permissive, right?